pgmax/fg/graph.py

```diff
@@ -53,48 +54,48 @@ class FactorGraph:
             this input, and the individual VariableGroups will need to be accessed by indexing.
     """
 
-    variables: Union[
-        Mapping[Any, groups.VariableGroup],
-        Sequence[groups.VariableGroup],
-        groups.VariableGroup,
-    ]
+    variables: Union[groups.VariableGroup, Sequence[groups.VariableGroup]]
 
     def __post_init__(self):
-        if isinstance(self.variables, groups.VariableGroup):
-            self._variable_group = self.variables
+        import time
+
+        start = time.time()
+        if isinstance(self.variables, (vgroup.NDVariableArray, vgroup.VariableDict)):
```

Comment:
Why isn't this just `isinstance(self.variables, groups.VariableGroup)`? If you do it this way when we add new variable groups in the future we would need to change this line